lib/constructs/persistence/data-pipeline.ts

```diff
@@ -173,17 +178,30 @@ export class SyllabusSyncPipeline extends AbstractDataPipeline {
     constructor(scope: cdk.Construct, id: string, props?: DataPipelineProps) {
         super(scope, id);
 
-        this.dataSource = props?.dataSource;
-
         this.dataWarehouse = new Table(this, 'dynamodb-syllabus-table', {
-            partitionKey: {name: "id", type: AttributeType.STRING},
+            partitionKey: {name: "school", type: AttributeType.STRING},
+            sortKey: {name: "id", type: AttributeType.STRING},
             billingMode: BillingMode.PROVISIONED,
             encryption: TableEncryption.DEFAULT,
             removalPolicy: cdk.RemovalPolicy.RETAIN,
             timeToLiveAttribute: "ttl",
-            tableName: "syllabus",
-            readCapacity: 5,
-            writeCapacity: 5
+            tableName: "waseda-syllabus",
+            readCapacity: 1,
+            writeCapacity: 1,
         });
+        //Use exsisting s3 bucket
+        this.dataSource = <Bucket>props?.dataSource;
```

Comment:
No need to use type assertion

Reply:
Is that mean that we need to change "abstract readonly dataSource?: Bucket" in AbstractDataPipeline to "abstract readonly dataSource: Bucket" cuz, if not do so, it may occur a type error.

Reply:
You can add exclamation mark after the field to indicate its not undefined.

Reply:
Fix